UnityClient/Assets/Scenes/Map/MapUiController.cs

```diff
@@ -102,4 +107,27 @@ public class MapUiController : MonoBehaviour {
     public void HideTooltip() {
         Tooltip.SetText(null, Vector3.zero);
     }
+
+    public void OnRestartAnswer(ushort cmd, int size, InPacket packet) {
+        if (packet is ZC.RESTART_ACK) {
+            var pkt = packet as ZC.RESTART_ACK;
+            if (pkt.type == 0) {
+                ChatBox.DisplayMessage(502, ChatMessageType.ERROR);
+            }
+            else {
+                // @todo ?
+                // clear StatusIcons
+                // clear ChatBox
+                // clear ShortCut
+                // clear PartyFriends
+                // clear renderers
+                OnRestart();
+            }
+        }
+    }
+
+    public void OnRestart() {
+        // @todo this keeps the entire UI on the screen
+        // SceneManager.LoadSceneAsync("CharSelectionScene");
```

Comment:
I could not find a solution for this issue in the project. When we return to the char select screen all the HUD remain

Reply:
The map ui is quite over complicated now... now that we load maps in a different way it might be easier to instantiate the UI later in time... you gotta hide the entire Ui when you unload this scene... grab the map ui object and get the canvas group component and turn its alpha down to 0... that should do the trick

Reply:
But with the alpha on 0 we could still click on the buttons, right?

Reply:
The `CanvasGroup` has another property which you can disable raycasts hits, so it would work